internal/types/path.go

```diff
@@ -17,17 +17,68 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+type Path struct {
+	s []string
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path slice copy.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Element returns path element specified by the index or empty string if index below 0 or greater than Path len.
+func (p Path) Element(index int) string {
+	if index < 0 || index > len(p.s) {
+		return ""
+	}
+	return p.s[index]
+}
+
+// Remove returns new path with removed path element specified by index.
+// If index below 0 or greater than Path len it would return the same value.
+func (p Path) Remove(index int) Path {
+	if index < 0 || index > len(p.s) {
+		return p
+	}
+	path := NewPath(p.s...)
+	path.s = append(path.s[:index], path.s[index+1:]...)
+	return path
+}
+
+// NewPath returns Path from a strings slice. Provided strings slice would be copied into a new Path.
+func NewPath(path ...string) Path {
```

Comment:
"Construct" functions should typically go before methods

Reply:
Fixed.